python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -32,46 +31,81 @@
 __all__ = ["OnDiskDataset", "preprocess_ondisk_dataset"]
 
 
-def preprocess_ondisk_dataset(input_config_path: str) -> str:
+def _copy_or_convert_data(
+    input_path,
+    output_path,
+    input_format,
+    output_format="numpy",
+    in_memory=True,
+):
+    """Copy or convert the data from input_path to output_path."""
+    os.makedirs(os.path.dirname(output_path), exist_ok=True)
+    if input_format == "numpy":
+        # If the original format is numpy, just copy the file.
+        shutil.copyfile(input_path, output_path)
+    else:
+        # If the original format is not numpy, convert it to numpy.
+        data = read_data(input_path, input_format, in_memory)
+        save_data(data, output_path, output_format)
+
+
+def preprocess_ondisk_dataset(dataset_dir: str) -> str:
     """Preprocess the on-disk dataset. Parse the input config file,
     load the data, and save the data in the format that GraphBolt supports.
 
     Parameters
     ----------
-    input_config_path : str
-        The path to the input config file.
+    dataset_dir : str
+        The path to the dataset directory.
 
     Returns
     -------
     output_config_path : str
         The path to the output config file.
     """
-    # 0. Load the input_config.
-    with open(input_config_path, "r") as f:
-        input_config = yaml.safe_load(f)
+    # Check if the dataset path is valid.
+    if not os.path.exists(dataset_dir):
+        raise RuntimeError(f"Invalid dataset path: {dataset_dir}")
+
+    # Check if the dataset_dir is a directory.
+    if not os.path.isdir(dataset_dir):
+        raise RuntimeError(
+            f"The dataset must be a directory. But got {dataset_dir}"
+        )
 
-    # If the input config does not contain the "graph" field, then we
-    # assume that the input config is already preprocessed.
-    if "graph" not in input_config:
-        print("The input config is already preprocessed.")
-        return input_config_path
+    # 0. Check if the dataset is already preprocessed.
+    if os.path.exists(os.path.join(dataset_dir, "preprocessed/metadata.yaml")):
+        print("The dataset is already preprocessed.")
+        return os.path.join(dataset_dir, "preprocessed/metadata.yaml")
 
     print("Start to preprocess the on-disk dataset.")
-    # Infer the dataset path from the input config path.
-    dataset_path = Path(os.path.dirname(input_config_path))
-    processed_dir_prefix = Path("preprocessed")
+    processed_dir_prefix = "preprocessed"
+    processed_dir_abs = os.path.join(dataset_dir, processed_dir_prefix)
```

Comment:
what is abs?

Reply:
short for `absolute`?

Reply:
Yes, but now we only need `processed_dir_prefix`.